File: output/ChatGPT/codenet/Java/C++/s356471394.cpp
#include <iostream>
#include <string>

using namespace std;

int N, M, D;
int num = 0;
char room[1000][1000];

void setWE() {
    bool canPut = true;

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < M - D + 1; j++) {
            canPut = true;

            for (int k = 0; k < D; k++) {
                if (room[i][j + k] == '#') {
                    canPut = false;
                    break;
                }
            }

            if (canPut == true) {
                num++;
            }
        }
    }
}

void setNS() {
    bool canPut = true;

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N - D + 1; j++) {
            canPut = true;

            for (int k = 0; k < D; k++) {
                if (room[j + k][i] == '#') {
                    canPut = false;
                    break;
                }
            }

            if (canPut == true) {
                num++;
            }
        }
    }
}

void inputRoom() {
    string str_room;

    for (int i = 0; i < N; i++) {
        cin >> str_room;
        for (int j = 0; j < M; j++) {
            room[i][j] = str_room[j];
        }
    }
}

int main() {
    cin >> N >> M >> D;

    inputRoom();

    if (M >= D) {
        setWE();
    }

    if (N >= D) {
        setNS();
    }

    cout << num << endl;

    return 0;
}

File: output/ChatGPT/codenet/Java/Python/s356471394.py
N, M, D = map(int, input().split())
room = []

for _ in range(N):
    room.append(list(input()))

num = 0

def setWE():
    global num
    canPut = True
    
    for i in range(N):
        for j in range(M - D + 1):
            canPut = True
            
            for k in range(D):
                if room[i][j + k] == '#':
                    canPut = False
                    break
            
            if canPut == True:
                num += 1

def setNS():
    global num
    canPut = True
    
    for i in range(M):
        for j in range(N - D + 1):
            canPut = True
            
            for k in range(D):
                if room[j + k][i] == '#':
                    canPut = False
                    break
            
            if canPut == True:
                num += 1

def inputRoom():
    global room
    for i in range(N):
        room.append(list(input()))

inputRoom()

if M >= D:
    setWE()

if N >= D:
    setNS()

print(num)